src/coreclr/jit/layout.cpp

```diff
@@ -400,7 +403,11 @@ void ClassLayout::InitializeGCPtrs(Compiler* compiler)
 // static
 bool ClassLayout::AreCompatible(const ClassLayout* layout1, const ClassLayout* layout2)
 {
-    assert((layout1 != nullptr) && (layout2 != nullptr));
+    if ((layout1 == nullptr) || (layout2 == nullptr))
+    {
+        return false;
+    }
```

Comment:
When does this happen now?

Reply:
It's a convenience feature (I can revert and null check at the call site if you think that'd be better).

The scenario when it happens is when we have a SIMD type local without a layout.

Reply:
Sounds reasonable to me